connectors/agent/config.py

```diff
@@ -0,0 +1,105 @@
+from connectors.config import add_defaults
+
+
+class ConnectorsAgentConfigurationWrapper:
+    """A wrapper that facilitates passing configuration from Agent to Connectors Service.
+
+    This class is responsible for:
+    - Storing in-memory configuration of Connectors Service running on Agent
+    - Transforming configuration reported by Agent to valid Connectors Service configuration
+    - Indicating that configuration has changed so that the user of the class can trigger the restart
+    """
+
+    def __init__(self):
+        """Inits the class.
+
+        There's default config that allows us to run connectors natively (see _force_allow_native flag),
+        when final configuration is reported these defaults will be merged with defaults from Connectors
+        Service config and specific config coming from Agent.
+        """
+        self._default_config = {
+            "_force_allow_native": True,
+            "native_service_types": [
+                "azure_blob_storage",
+                "box",
+                "confluence",
+                "dropbox",
+                "github",
+                "gmail",
+                "google_cloud_storage",
+                "google_drive",
+                "jira",
+                "mongodb",
+                "mssql",
+                "mysql",
+                "notion",
+                "onedrive",
+                "oracle",
+                "outlook",
+                "network_drive",
+                "postgresql",
+                "s3",
+                "salesforce",
+                "servicenow",
+                "sharepoint_online",
+                "slack",
+                "microsoft_teams",
+                "zoom",
+            ],
+        }
+
+        self.specific_config = {}
+
+    def try_update(self, source):
+        """Try update the configuration and see if it changed.
+
+        This method takes the check-in event coming from Agent and checks if config needs an update.
+
+        If update is needed, configuration is updated and method returns True. If no update is needed
+        the method returns False.
+        """
+
+        # TODO: find a good link to what this object is.
```

Comment:
Trying to be helpful, so digged into proto def, https://github.com/elastic/elastic-agent-client/blob/main/elastic-agent-client.proto#L168 

But it seems like `source` is a an arbitrary struct hehe (so we have to look somewhere else for the actual definition) and take a leap of faith now

Reply:
The way I interrogated this during my POC was to just add logging. I'd get a checkin event, deserialize it, re-serialize it as json, then dump it in the logs.